### PR TITLE
Zigbee protocol support

### PR DESCRIPTION
**Pull request by P-R-O-C-H-Y**
## Description of Change
Adds menu in tools for selecting Zigbee mode to include proper libraries.

## Tests scenarios
Tested with reworked esp-idf zigbee on/off light example with manually build libraries by lib-builder with this [PR](https://github.com/espressif/esp32-arduino-lib-builder/pull/146) applied .

## Related links
Related #8807

### CODE REVIEW

File: boards.txt [hunk 1]
```diff
@@ -17,6 +17,7 @@ menu.EventsCore=Events Run On
 menu.MemoryType=Memory Type
 menu.EraseFlash=Erase All Flash Before Sketch Upload
 menu.JTAGAdapter=JTAG Adapter
+menu.Zigbee=Zigbee Mode
```
Comment (me-no-dev) [inline boards.txt:20]: please rename the menu to `ZigbeeMode`. This is a bit too generic

File: boards.txt [hunk 2]
```diff
@@ -187,6 +194,20 @@ esp32h2.menu.EraseFlash.none.upload.erase_cmd=
 esp32h2.menu.EraseFlash.all=Enabled
 esp32h2.menu.EraseFlash.all.upload.erase_cmd=-e
 
+esp32h2.menu.Zigbee.default=Disabled
+esp32h2.menu.Zigbee.default.build.zigbee_mode=
+esp32h2.menu.Zigbee.default.build.extra_libs=
+esp32h2.menu.Zigbee.ed=Zigbee ED (end device)
+esp32h2.menu.Zigbee.ed.build.zigbee_mode=-DZIGBEE_MODE_ED
+esp32h2.menu.Zigbee.ed.build.extra_libs=-lesp_zb_api_ed -lesp_zb_cli_command -lzboss_stack.ed.trace -lzboss_stack.ed -lzboss_port
```
Comment (me-no-dev) [inline boards.txt:202]: let's allow `build.extra_libs` to be available for something else that would be use-case specific. Add `build.zigbee_libs` instead. Makes it more consistent and prevents some other feature to overwrite the variable.

File: boards.txt (hunk 1 above)
Comment (P-R-O-C-H-Y) [reply boards.txt:20]: Done.

File: boards.txt (hunk 2 above)
Comment (P-R-O-C-H-Y) [reply boards.txt:202]: Done, good catch as that could collide with the PSRAM extra_libs. Renamed as suggested

**Pull request merged**